src/main.ts

```diff
@@ -13,88 +11,74 @@ import GitHubRelease, {
   IGitHubReleaseOptions
 } from './github-release';
 
+import { AsyncSeriesBailHook, AsyncSeriesHook, SyncHook } from 'tapable';
 import init from './init';
+import NpmPlugin from './plugins/npm/npm';
+import SEMVER from './semver';
 import execPromise from './utils/exec-promise';
+import getGitHubToken from './utils/github-token';
 import createLog from './utils/logger';
 
-const readFile = promisify(fs.readFile);
+interface IAuthor {
+  name: string;
+  email: string;
+}
+
+interface IRepository {
+  owner: string;
+  repo: string;
+  token?: string;
+}
 
-const calcGreaterVersion = (
-  packageVersion: string,
-  monorepoVersion: string,
-  latestVersion: string
-) => {
-  const localVersion = monorepoVersion || packageVersion;
+export interface IAutoHooks {
+  beforeRun: SyncHook<[IGitHubReleaseOptions]>;
+  getAuthor: AsyncSeriesBailHook<[], IAuthor>;
+  getPreviousVersion: AsyncSeriesBailHook<
+    [(release: string) => string],
+    string
+  >;
+  getRepository: AsyncSeriesBailHook<[], IRepository>;
+  publish: AsyncSeriesHook<[SEMVER]>;
+}
 
-  return localVersion > latestVersion ? localVersion : latestVersion;
-};
+async function getRepo(args: ArgsType, hooks: IAutoHooks) {
+  hooks.getRepository.tap('None', () => args as IRepository);
 
-function isMonorepo() {
-  return fs.existsSync('lerna.json');
+  return hooks.getRepository.promise();
 }
 
 async function getCurrentVersion(
   prefixRelease: (release: string) => string,
   lastRelease: string,
-  veryVerbose: signale.Signale<signale.DefaultMethods>
+  veryVerbose: signale.Signale<signale.DefaultMethods>,
+  hooks: IAutoHooks
 ) {
-  let lastVersion;
-  let packageVersion = '';
-  let monorepoVersion = '';
-
-  if (isMonorepo()) {
-    monorepoVersion = prefixRelease(
-      JSON.parse(await readFile('lerna.json', 'utf-8')).version
+  hooks.getPreviousVersion.tap('None', () => {
+    veryVerbose.info(
+      'No previous release found, using 0.0.0 as previous version.'
     );
-  }
+    return prefixRelease('0.0.0');
+  });
 
-  if (fs.existsSync('package.json')) {
-    packageVersion = prefixRelease(
-      JSON.parse(await readFile('package.json', 'utf-8')).version
-    );
-  }
+  const lastVersion = await hooks.getPreviousVersion.promise(prefixRelease);
 
-  if (lastRelease.match(/\d+\.\d+\.\d+/)) {
+  if (lastRelease.match(/\d+\.\d+\.\d+/) && lastRelease > lastVersion) {
```

Comment:
Aren't they both strings? Not sure `>` is doing what you want it to do

Reply:
using sember.gt now! good catch